src/main/webapp/script.js

```diff
@@ -0,0 +1,135 @@
+
+  addObjects();
+  gui();
+}
+
+/**
+ * Creates all the objects and adds them to the scene.
+*/
+function addObjects() {
+  let geometry = new THREE.PlaneGeometry(5, 25, 1);
+  let material =
+      new THREE.MeshBasicMaterial({color: 'pink', side: THREE.DoubleSide});
+  // The plane is a place holder for the 2D localized map image
+  const plane = new THREE.Mesh(geometry, material);
+  plane.rotation.x = Math.PI / 2;
+  plane.position.z = -10;
+  scene.add(plane);
+
+  // Fake Pose Data
+  material = new THREE.LineBasicMaterial({color: 'red'});
+  for (let increment = 0; increment < numOfPoints; increment++) {
+    points.push(new THREE.Vector3(
+        Math.pow(-1, increment), .20, -increment)); // GPS points
+
+    const geometry = new THREE.CylinderGeometry(.02, .02, .5);
+
+    // The cylinder represents the orientation of each data point
+    const cylinder = new THREE.Mesh(geometry, material);
+    cylinder.rotateX(THREE.Math.degToRad(-90));
+    cylinder.position.set(0, 0, -.25);
+    scene.add(cylinder); // Orientation indicator
+
+    // The pivot sphere represents the position of each data point
+    const pivot = new THREE.Group();
+    pivot.position.set(Math.pow(-1, increment), .25, -increment);
+    scene.add(pivot);
+    pivot.add(cylinder);
+    const pivotSphereGeo =
+      new THREE.SphereGeometry(.03); // Small sphere indicating pivot point
+    const pivotSphere = new THREE.Mesh(pivotSphereGeo);
+    pivotSphere.position.set(Math.pow(-1, increment), .25, -increment);
+    scene.add(pivotSphere);
+
+    rotationData.set(increment, {origin: pivot, mesh: cylinder});
+  }
+  geometry = new THREE.BufferGeometry().setFromPoints(points);
+  material = new THREE.LineBasicMaterial({color: 'blue'});
+  const line = new THREE.Line(geometry, material);
+  scene.add(line);
+}
+
+/**
+ * Creates a GUI for manipulating the orientation of each data point
+ */
+function gui() {
+  const gui = new GUI();
+  let params = {clipIntersection: true, Yaw: 0, showHelpers: false};
+  gui.add(params, 'Yaw', -360, 360)
+      .step(.1)
+      .name('Yaw (degrees)')
+      .onChange(function(radians) {
+        for (let i = 0; i < rotationData.size; i++) {
+          rotationData.get(i).origin.rotation.z = THREE.Math.degToRad(radians);
+        }
+      });
+
+  params = {clipIntersection: true, Pitch: 0, showHelpers: false};
+  gui.add(params, 'Pitch', -360, 360)
+      .step(.1)
+      .name('Pitch (degrees)')
+      .onChange(function(radians) {
+        for (let i = 0; i < rotationData.size; i++) {
+          rotationData.get(i).origin.rotation.x = THREE.Math.degToRad(radians);
+        }
+      });
+
+  params = {clipIntersection: true, Roll: 0, showHelpers: false};
+  gui.add(params, 'Roll', -360, 360)
+      .step(.1)
+      .name('Roll (degrees)')
+      .onChange(function(radians) {
+        for (let i = 0; i < rotationData.size; i++) {
+          rotationData.get(i).origin.rotation.y = THREE.Math.degToRad(radians);
+        }
+      });
```

Comment:
These three blocks for roll, pitch, and yaw are very similar. Can you factor out the common behavior in order to share code between them? That will also make it easier to scan for the differences between them, because the things that vary will be function arguments.

Reply:
LGTM. In case you're interested, another way you could do this is have `rotateAxis(axis)` return a function that accepts `radians`; so `rotateAxis` is a higher order function.

https://en.wikipedia.org/wiki/Higher-order_function